### PR TITLE
Reflection

### PR DESCRIPTION
**Pull request by BenHSmith13**
Serializes and deserializes qbank data

### CODE REVIEW

File: client/js/_author/reducers/items.js [hunk 1]
```diff
@@ -1,42 +1,56 @@
...
-  const newItem = _.cloneDeep(item);
-
-  if (!newItem.question) {
-    newItem.question = {
-      choices: {},
-    };
-    return newItem;
-  }
-
+function deserializeChoices(choices, answers) {
```
Comment (dspencer001) [inline client/js/_author/reducers/items.js]: I think in the future we should figure out how to put all of this in the middleware. For now, merging this is probably more important.

File: client/js/middleware/serializers/qbank/multiple_choice_serializer.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import _              from 'lodash';
```
Comment (dspencer001) [inline client/js/middleware/serializers/qbank/multiple_choice_serializer.js]: We should probably add a comment somewhere (not sure where, maybe here or in the reducer) showing what the end object looks like.

Comment (BenHSmith13) [inline client/js/middleware/serializers/qbank/multiple_choice_serializer.js]: Which end? And thats kind of what like 11 in this file does.

File: client/js/_author/reducers/items.js (hunk 1 above)
Comment (BenHSmith13) [inline client/js/_author/reducers/items.js]: agreed

File: client/js/middleware/serializers/qbank/multiple_choice_serializer.js (hunk 2 above)
Comment (dspencer001) [inline client/js/middleware/serializers/qbank/multiple_choice_serializer.js]: For whatever reason I got the serializer and deserializes mixed up. We probably need comments on both though, although then maintaining that becomes a royal pain. I'm thinking sorta similar to the canvas docs, just so you can be reasonably sure what the object is going to look like, without going and looking at the actual code that generates it.

Comment (dspencer001) [inline client/js/middleware/serializers/qbank/multiple_choice_serializer.js]: That's less important for the qbank side though.

File: client/js/_author/reducers/items.js [hunk 3]
```diff
@@ -1,42 +1,56 @@
...
       }
     });
   });
-  newItem.question.choices = newChoices;
-  return newItem;
+  return newChoices;
+}
+
+// The implementation of function is Q-Bank specific
+function deserializeItem(item) {
```
Comment (dspencer001) [inline client/js/_author/reducers/items.js]: A lot of this seems very multiple choice specific. How does this handle things that don't have choices?

**Pull request closed without merging**